Source/WebCore/platform/cocoa/ParentalControlsContentFilter.mm

```diff
@@ -80,26 +90,19 @@ static inline bool canHandleResponse(const ResourceResponse& response)
 
 void ParentalControlsContentFilter::responseReceived(const ResourceResponse& response)
 {
-#if HAVE(WEBCONTENTRESTRICTIONS)
-    bool filterEnabled = enabled(m_usesWebContentRestrictions);
-#else
-    bool filterEnabled = enabled();
-#endif
-    if (!canHandleResponse(response) || !filterEnabled) {
+    if (!canHandleResponse(response) || !enabled()) {
         m_state = State::Allowed;
         return;
     }
 
 #if HAVE(WEBCONTENTRESTRICTIONS)
-    if (m_usesWebContentRestrictions) {
-        ASSERT(!m_wcrBrowserEngineClient);
-        m_wcrBrowserEngineClient = adoptNS([PAL::allocWCRBrowserEngineClientInstance() init]);
+    if (RefPtr urlFilter = m_urlFilter) {
         m_evaluatedURL = response.url();
         m_state = State::Filtering;
-        [m_wcrBrowserEngineClient evaluateURL:*m_evaluatedURL withCompletion:[weakThis = WeakPtr { *this }](BOOL shouldBlock, NSData *replacementData) {
+        urlFilter->isURLAllowed(*m_evaluatedURL, [weakThis = WeakPtr { *this }](bool allowed, auto&& replacementData) {
             if (CheckedPtr checkedThis = weakThis.get())
-                checkedThis->updateFilterState(shouldBlock, replacementData);
-        }];
+                checkedThis->updateFilterState(!allowed, WTFMove(replacementData));
```

Comment:
Ideally, `updateFilterState()` would take in a enum class. The switching back and forth between `allowed` (which should be named `isAllowed` BTW) and `shouldBlock` is error-prone.